pdf/pdfjob.cpp

```diff
@@ -41,17 +50,130 @@ void LoadDocumentJob::run()
     }
 }
 
-UnLockDocumentJob::UnLockDocumentJob(const QString &password)
-    : PDFJob(PDFJob::UnLockDocumentJob), m_password(password)
+class DbWorker
+{
+public:
+    DbWorker(const QString &path);
+    ~DbWorker();
+
+    QByteArray load() const;
+    void clear() const;
+    void store(const QByteArray &password) const;
+
+private:
+    QString m_path;
+};
+
+DbWorker::DbWorker(const QString &path)
+    : m_path(path)
+{
+    if (!m_path.isEmpty() && !QSqlDatabase::contains()) {
+        QSqlDatabase db = QSqlDatabase::addDatabase("QSQLITE");
+        db.setDatabaseName(QStandardPaths::writableLocation(QStandardPaths::DataLocation)
```

Comment:
DataLocation is deprecated, replaced by AppDataLocation.

Reply:
Good point, fixed. Thank you.